patches/api/0442-Add-player-idle-duration-API.patch

```diff
@@ -0,0 +1,49 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: booky10 <boooky10@gmail.com>
+Date: Sat, 14 Oct 2023 03:11:11 +0200
+Subject: [PATCH] Add player idle duration API
+
+Implements API for getting, setting and resetting a player's idle duration.
+
+diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
+index 20fa1024f9ad8f478a347be5c554b5e45b398a1c..2fc042b8ffe610a7ec431aad3c1d3ba407f05d40 100644
+--- a/src/main/java/org/bukkit/entity/Player.java
++++ b/src/main/java/org/bukkit/entity/Player.java
+@@ -3274,6 +3274,37 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
+     void increaseWardenWarningLevel();
+     // Paper end
+ 
++    // Paper start
++    /**
++     * The idle duration is reset when the player
++     * sends specific action packets.
++     * <p>
++     * After the idle duration exceeds {@link org.bukkit.Bukkit#getIdleTimeout()}, the
++     * player will be kicked for {@link org.bukkit.event.player.PlayerKickEvent.Cause#IDLING}.
++     *
++     * @return the current idle duration of this player
++     */
++    @NotNull Duration getIdleDuration();
++
++    /**
++     * After the idle duration exceeds {@link org.bukkit.Bukkit#getIdleTimeout()}, the
++     * player will be kicked for {@link org.bukkit.event.player.PlayerKickEvent.Cause#IDLING}.
++     *
++     * @param duration the new idle duration of this player
++     */
++    void setIdleDuration(@NotNull Duration duration);
```

Comment:
What is a potential usecase for setting the idle duration here?
Resetting I certainly get, but what would a manual setting be useful for ?

Reply:
I've mainly used the get and reset method before, but when creating this PR it seemed odd to have a getter but no setter. I see no reason not to add it.

One possible use case for this would be to persist the idle duration between reconnects, but that seems like a very niche use case to me.

Reply:
Well I am more concerned about the fact that, at least rn, Duration is a rather wide spectrum of numbers possible to be passed.

Passing some terribly high duration here would quickly end up with the player having an idle duration that makes 0 sense, e.g. longer than they have been online or longer than the server has been up.

Reply:
Yeah, makes sense, I'm gonna remove that then